src/Payments.sol

```diff
@@ -390,19 +390,47 @@ contract Payments is
             rail.from
         ][rail.operator];
 
-        // Ensure account lockup is fully settled up to the current epoch
-        uint256 lockupSettledUpto = settleAccountLockup(payer);
         require(
-            lockupSettledUpto == block.number,
-            "cannot modify active rail lockup: client funds insufficient for current account lockup settlement"
+            !isRailInDebt(rail, payer),
+            "cannot modify rail lockup: rail is in debt"
         );
 
-        // Calculate current (old) lockup - for active rails this is straightforward
+        // Settle account lockup as much as possible
+        uint256 lockupSettledUpto = settleAccountLockup(payer);
+
+        // Only require full settlement if increasing period or fixed lockup
+        if (period > rail.lockupPeriod || lockupFixed > rail.lockupFixed) {
+            require(
+                lockupSettledUpto == block.number,
+                "cannot increase lockup: client funds insufficient for current account lockup settlement"
+            );
+        } else {
+            // When reducing period, ensure we still cover all unsettled epochs
+            if (period < rail.lockupPeriod) {
+                require(
+                    payer.lockupLastSettledAt + period >= block.number,
+                    "cannot reduce lockup period below what's needed for unsettled epochs"
+                );
+            }
+        }
+
+        // Calculate effective lockup period for the old period
+        uint256 oldEffectiveLockupPeriod = rail.lockupPeriod -
+            (block.number - payer.lockupLastSettledAt);
+
+        // Calculate effective lockup period for the updated period
+        uint256 newEffectiveLockupPeriod = period -
+            (block.number - payer.lockupLastSettledAt);
+
```

Comment:
What happens if we reduce the fixed lockup but the rail is way in debt? I think these statements will revert.